src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,105 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private T[] values;
+    private int size = 0;
+
+    public ArrayList() {
+        values = (T[]) new Object[DEFAULT_SIZE];
+    }
+
+    public void resizeArray() {
+        int newSize = (size + (size >> 1));
+        values = Arrays.copyOf(values, newSize);
```

Comment:
[checklist](https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/array-list.html#use-systemarraycopy-to-move-your-array-elements): Use [System.arraycopy()](https://docs.oracle.com/javase/8/docs/api/java/lang/System.html#arraycopy-java.lang.Object-int-java.lang.Object-int-int-) to move your array elements